scripts/reconcile-task-projections.py

```
@@ -0,0 +1,678 @@
+    lease_doc = packet.load(lease_path, dict if kind == "worker" else list, required=not legacy)
+    leases = packet.rows([lease_doc] if kind == "worker" and lease_doc is not None else lease_doc, lease_path)
+    lease_index = packet.index(leases, lease_path)
+    for lease in leases:
+        joined(lease, lease_path)
+        if lease.get("schema") != "ub-review.resource_lease.v1":
+            packet.issue("unsupported_lease_schema", lease_path, lease.get("id", ""))
+        if lease.get("status") == "granted" and lease.get("consumer") not in receipt_index:
+            packet.issue("granted_lease_without_receipt", lease_path, lease.get("id", ""))
+    for proof in proofs:
+        if any(c.get("status") in {"passed", "failed", "timed_out"} for c in proof.get("commands", [])):
+            if not legacy and not any(l.get("consumer") == proof.get("id") and l.get("status") == "granted" for l in leases):
```

Ruff (0.16.3)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)